tests/test_main.py

```diff
@@ -17,262 +15,205 @@ def test_get_words_incorrect_username(self):
     def test_get_words_correct_username(self):
         self.assertEqual(get_words("@user I'm a message", 'user'), ['@user', "I'm", 'a', 'message'])
 
-    @patch('homu.main.PullReqState')
-    @patch('homu.main.get_portal_turret_dialog', return_value='message')
-    def test_still_here(self, mock_message, MockPullReqState):
-        state = MockPullReqState()
-        still_here(state)
-        state.add_comment.assert_called_once_with(':cake: message\n\n![](https://cloud.githubusercontent.com/assets/1617736/22222924/c07b2a1c-e16d-11e6-91b3-ac659550585c.png)')
-
-    @patch('homu.main.PullReqState')
-    def test_hello_or_ping(self, MockPullReqState):
-        state = MockPullReqState()
-        hello_or_ping(state)
-        state.add_comment.assert_called_once_with(":sleepy: I'm awake I'm awake")
-
-    @patch('homu.main.PullReqState')
-    def test_treeclosed_negative(self, MockPullReqState):
-        state = MockPullReqState()
-        treeclosed_negative(state)
-        state.change_treeclosed.assert_called_once_with(-1)
-        state.save.assert_called_once_with()
+    def test_sha_or_blank_return_sha(self):
+        self.assertEqual(sha_or_blank('f5d42200481'), 'f5d42200481')
 
-    @patch('homu.main.PullReqState')
-    def test_set_treeclosed(self, MockPullReqState):
-        state = MockPullReqState()
-        set_treeclosed(state, 'treeclosed=123')
-        state.change_treeclosed.assert_called_once_with(123)
-        state.save.assert_called_once_with()
+    def test_sha_or_blank_return_blank(self):
+        self.assertEqual(sha_or_blank('f5d@12'), '')
 
+    @patch('homu.main.get_words', return_value=["@bot", "are", "you", "still", "there?"])
+    @patch('homu.main.verify_auth', return_value=True)
     @patch('homu.main.PullReqState')
-    def test_rollup_positive(self, MockPullReqState):
+    @patch('homu.action.Action.still_here')
+    def test_parse_commands_still_here_realtime(self, mock_still_here, MockPullReqState, mock_auth, mock_words):
         state = MockPullReqState()
-        rollup(state, 'rollup')
-        self.assertTrue(state.rollup)
-        state.save.assert_called_once_with()
+        self.assertFalse(parse_commands({}, '', '', {}, state, '', '', [], realtime=True))
```

Comment:
Let's make the tests a bit easier to understand by creating a helper that calls parse_commands with a default set of values so that we can just pass in keyword arguments that are different than the defaults.